src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -158,7 +159,50 @@ void VehicleInfoPlugin::DeleteSubscriptions(
     }
   }
 }
+
+void VehicleInfoPlugin::SelectVehicleDataForUnsubscribe(
+    const app_mngr::ApplicationSharedPtr& app,
+    application_manager::VehicleInfoSubscriptions& vehicle_data_for_unscribe) {
+  auto vehicle_extension = VehicleInfoAppExtension::ExtractVIExtension(*app);
+  vehicle_data_for_unscribe = vehicle_extension.Subscriptions();
+  application_manager::ApplicationSet applications =
+      application_manager_->applications().GetData();
+  if (applications.empty()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Other applications not exists. All existing vehicle data "
```

Comment:
@IGapchuk you make an assumption that this method called after the app application was erased from the application list. This assumption is correct, but got user that will call this function further it is no obvious. For example, somebody will call this function on policy update if the application was disallowed. In that case, `applications()` won't be empty, and vehicle_data_collector will be called for each subscription, and for each application (including the app) because you have no check that `SelectVehicleDataForUnsubscribe::app` and `vehicle_data_collector::app` is equal we will have the situation that any data won't be elected ( everything will be erased).
So I would not make an assumption that `application()` is empty if the app is the only one existing app. I would better putt additional check inside vehicle_data_collector, and remove check if `applications()` is empty

Reply:
Done in e8b297ab1e2a8dee6bd07fdf2e64d38d1dd62b33.